src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlSecurityUtility.cs

```diff
@@ -268,13 +271,19 @@ internal static void DecryptSymmetricKey(SqlTceCipherInfoEntry sqlTceCipherInfoE
             sqlClientSymmetricKey = null;
             encryptionkeyInfoChosen = null;
             Exception lastException = null;
-            SqlSymmetricKeyCache cache = SqlSymmetricKeyCache.GetInstance();
+            SqlSymmetricKeyCache globalCekCache = SqlSymmetricKeyCache.GetInstance();
 
             foreach (SqlEncryptionKeyInfo keyInfo in sqlTceCipherInfoEntry.ColumnEncryptionKeyValues)
             {
                 try
                 {
-                    if (cache.GetKey(keyInfo, out sqlClientSymmetricKey, connection))
+                    if (InstanceLevelProvidersAreRegistered(connection, command) &&
+                        TryGetKeyFromLocalCache(keyInfo, out sqlClientSymmetricKey, connection, command))
```

Comment:
3 Issues:
1. We never want to fall through to the `else if` if InstanceLevelProvidersAreRegistered returns true.
2. TryGetKeyFromLocalCache is not a true try/get function. It only returns true and throws an exception in all other cases. 
3. TryGetKeyFromLocalCache seems mis-named. It's not getting the key from a local cache. It's getting the key from the provider.

I would change the function to
`private static SqlClientSymmetricKey GetKeyFromLocalProviders(SqlEncryptionKeyInfo keyInfo, SqlConnection connection, SqlCommand command)`
and then a simple call inside the if block:
`sqlClientSymmetricKey = GetKeyFromLocalProviders(keyInfo, connection, command);`
so the logic and intention here is clear.

I realize globalCekCache.GetKey behaves the same way (meaningless bool return with an out variable), but it doesn't seem like a good pattern to replicate since it can cause confusion about the intention of the logic. If you want to refactor that function, now would be a good time to do it since you are already touching it, but not a requirement for this PR.